### PR TITLE
Fix invalid SQL for timestamp comparison in psql

### PR DESCRIPTION
**Pull request by sbelak**
Fixes a bug in cache results cleanup when using Postgres for application database.

### CODE REVIEW

**Review by senior — COMMENTED** [review]
Would be good to get a test on this and have it run against PG (i.e. [ci drivers])

File: test/metabase/task/cleanup_temorary_computation_job_results_test.clj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            [toucan.util.test :as tt]))
+
+(tt/expect-with-temp
+ [ComputationJob       [{job-id :id} {:type   :simple-job
+                                      :status :done}]
+  ComputationJobResult [{result-id :id} {:job_id     job-id
+                                         :permanence :temporary
+                                         :payload    1}]]
+  true
+  (let [c1 (count (ComputationJobResult))]
```
Comment (camsaul) [inline test/metabase/task/cleanup_temorary_computation_job_results_test.clj]: you can do `(db/count ComputationJobResult)` instead. That produces a SQL query like `SELECT COUNT(*) ...` whereas doing `count` in Clojure-land pulls down every object from the DB with `SELECT *` and *then* counts them.

**Comment by camsaul** [comment]
Any reason we're still sitting on this?

File: test/metabase/task/cleanup_temorary_computation_job_results_test.clj [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            [toucan.db :as db]
+            [toucan.util.test :as tt]))
+
+(tt/expect-with-temp
+ [ComputationJob       [{job-id :id} {:type   :simple-job
+                                      :status :done}]
+  ComputationJobResult [{result-id :id} {:job_id     job-id
+                                         :permanence :temporary
+                                         :payload    1}]]
+  true
```
Comment (camsaul) [inline test/metabase/task/cleanup_temorary_computation_job_results_test.clj]: You shouldn't use `expect-with-temp` if the `expected` form doesn't need it. Just do

```clojure
(expect
  (tt/with-temp* ...))
```

(`expect` with one are is the same as `(expect true ...)`

**Comment by camsaul** [comment]
I went ahead and fixed the last couple things here because the error messages were making it hard for me to debug something

**Pull request merged**